config/mapflags.go

```diff
@@ -0,0 +1,59 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// mapFlags are generic string key-value pair flags.
+// Use when option keys are not predetermined.
+type mapFlags struct {
+	values map[string]string
+}
+
+func (m mapFlags) String() string {
+	var pairs []string
+
+	for k, v := range m.values {
+		pairs = append(pairs, fmt.Sprint(k, "=", v))
+	}
+
+	return strings.Join(pairs, "'")
+}
+
+func (m *mapFlags) Set(value string) error {
+	if m == nil {
+		return nil
+	}
+
+	m.values = make(map[string]string)
+
+	vs := strings.Split(value, ",")
+	for _, vi := range vs {
+		kv := strings.Split(vi, "=")
+
+		k := strings.TrimSpace(kv[0])
+		v := strings.TrimSpace(kv[1])
+
+		if len(kv) != 2 || k == "" || v == "" {
```

Comment:
The length is checked after slice is addressed, also https://golang.org/pkg/strings/#SplitN might be useful

Reply:
Fixed with additional unit tests as well.